test/src/subnet/main.cpp

```diff
@@ -0,0 +1,141 @@
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include <string>
+#include <cadmium\modeling\coupling.hpp>
+#include <cadmium\modeling\ports.hpp>
+#include <cadmium\modeling\dynamic_model_translator.hpp>
+#include <cadmium\concept\coupled_model_assert.hpp>
+#include <cadmium\modeling\dynamic_coupled.hpp>
+#include <cadmium\modeling\dynamic_atomic.hpp>
+#include <cadmium\engine\pdevs_dynamic_runner.hpp>
+#include <cadmium\logger\tuple_to_ostream.hpp>
+#include <cadmium\logger\common_loggers.hpp>
+
+#include "..\..\..\lib\vendor\NDTime.hpp"
+#include "..\..\..\lib\vendor\iestream.hpp"
+#include "..\..\..\include\data_structures\message.hpp"
+#include "..\..\..\include\atomics\subnet_cadmium.hpp"
+
+using namespace std;
+
+using hclock=chrono::high_resolution_clock;
+using TIME = NDTime;
+
+/***** SETING INPUT PORTS FOR COUPLEDs *****/
+struct inp_in: public cadmium::in_port<Message_t> {
+};
+
+/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
+struct outp_out: public cadmium::out_port<Message_t> {
+};
+
+/********************************************/
+/****** APPLICATION GENERATOR *******************/
+/********************************************/
+template<typename T>
+class ApplicationGen: public iestream_input<Message_t, T> {
+public:
```

Comment:
check indentation.... should be like following:
class ApplicationGen: public iestream_input<Message_t, T> {
    public:
    ApplicationGen() = default;
    ApplicationGen(const char *file_path) :
    iestream_input<Message_t, T>(file_path) {
    }
};